### DISCUSSION TITLE
希望github自动集成流程能够更智能一点

### DISCUSSION THREAD

**Discussion by xzl01** [linuxdeepin/developer-center#5404] [Features Request & Ideas | 特性请求 & 头脑风暴]
- [x] ~~支持topic 同时集成（集成任意一个topic内软件包，整个topic全部集成）~~ 代码实现在 https://github.com/hudeng-go/test-infra/pull/5 ，目前支持`/integr-topic $projectName`命令一键集成topic关联的project（因github 的api目前只开放classic project相关的操作，因此该命令也只能支持该类型的project）中所有的pr。
 - [ ] 支持自动集成后人为干预，手动干预集成软件包内容（在自动集成后，可以人为修改集成内的软件包，增加或删除）

**Comment by hudeng-go** [comment]
计划使用prow webhook机器人插件实现，topic的管理在github平台上借用project进行关联管理，然后实现一键topic集成和状态同步，该方案使用过程中不支持跨组织的项目集成（目前有deepin-community和linuxdeepin组织），不过个人认为不需要也不应该允许跨组织的集成，本身这个两个组织的项目更新体现到软件包仓库里是分成dde组件和系统组件的，分组件的目的之一就是实现隔离更新，因此跨组织的topic集成建议目前不提倡！！！

ps： 插件名字就叫deepin-topic插件？

**Reply by BLumia** [reply]
> ps： 插件名字就叫deepin-topic插件？

如果涉及仓库名，叫 deepin-prow-plugin-topic？能体现是 prow 的插件。不涉及的话就无所谓了。

**Reply by hudeng-go** [reply]
本身插件就在[prow](https://github.com/hudeng-go/test-infra)仓库当中，所以目前不涉及仓库名。
